src/producer.cpp

```diff
@@ -109,7 +109,7 @@ Producer::ckDataGen(const Policy& accessPolicy, std::shared_ptr<Data> dataTempla
     auto ckData = std::move(dataTemplate);
     ckData->setName(ckDataName);
     ckData->setContent(contentKey->makeCKContent());
-    m_keyChain.sign(*ckData, signingByCertificate(m_cert));
+    m_keyChain.sign(*ckData, signingWithSha256());
```

Comment:
This doesn't help anything. If we want to provide flexibility in signing, we should provide a signingInfo as an input parameter.
See [ndn-svs](https://github.com/named-data/ndn-svs/blob/master/examples/chat-pubsub.cpp#L45).